common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -663,6 +661,11 @@ def _compute_metadata_inheritance_tree(self, course_id):
             # manually pick it apart b/c the db has tag and we want as_published revision regardless
             location = as_published(Location._from_deprecated_son(result['_id'], course_id.run))
 
+            # TODO This level of detail should not be here. It is necessary because the display name for video
+            # modules defaults to "Video" if one is not provided. We should find a better method of setting this value.
+            if result['_id']['category'] == 'video' and not result['metadata'].get('display_name'):
+                result['metadata']['display_name'] = 'Video'
```

Comment:
Damn. I was really hoping that `display_name` would be something safe to pull, if nothing else. I guess this is a good warning on the abstractions we're leaking through in this -- as you point out, the VideoModule should have a say here.